### PR TITLE
add the q3 answers for part 1

### PR DESCRIPTION
**Pull request by IdanTo1**
Signed-off-by: Idan Tomer <idan.tomer2@gmail.com>

### CODE REVIEW

File: part1-q3-ans.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+<var-ref> ::= an identifier token
+<var-decl> ::= an identifier token
+<sexp> ::= symbol | number | bool | string |
+(<sexp>+ . <sexp>) | ( <sexp>* ) ##### L3
+
+3.2
+to remove the use of ValueToLitExp, we added to the cexp type the sexp value, so that when getting a closure, we can call substitute imediatlly, 
+without the need to cast it back to CExp type first.
+to make the change complete, we will pass to substitute the args of the closure without casting them to Cexp, and in the substitute function,
+in the "sub" function, we'll add handling to Sexp, extracting it's nameץ
```
Comment (omerporze) [inline part1-q3-ans.txt:34]: what is the ץ at the end?

File: part1-q3-ans.txt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+3.3
+Change types option has the big advantage of saving the doubled transitions.
+in the ValueToLitExp approach, we process a Value expression, that was infered from CExp, and then to evaluate the Closure, we cast it back to CExp,
+just to rename the vars inside the closure to prevent shadowind. In this new approach we're able to extract the name from the Value directly, and there's
```
Comment (omerporze) [inline part1-q3-ans.txt:40]: shadowind -> shadowing

File: part1-q3-ans.txt (hunk 1 above)
Comment (IdanTo1) [reply part1-q3-ans.txt]: should have been a dot, fixed

File: part1-q3-ans.txt (hunk 2 above)
Comment (IdanTo1) [reply part1-q3-ans.txt]: fixed

**Pull request merged**